docs/how-to-integrate-a-frontend/how-to-integrate-dotwallet.md

```diff
@@ -0,0 +1,196 @@
+    |response_type |YES | Fill in the fixed value : `code` |
+    |scope | YES | Authorization scope. The list of permissions that the user agrees to authorize. These permissions are required for certain API endpoints. Needs to be url_encoded. Use spaces to separate multiple permissions. For a list of currently supported scope permissions, please check the scope list below|
+
+2. Redirect the user to the URI constructed in **1**
+   
+    After clicking the link, the user will be directed to the DotWallet authorization page. DotWallet will ask the user to log in, and then ask whether they agree to authorize the application for the listed permission scopes.
+
+3. Receive the `code` through the callback uri.
+
+    After the user agrees to authorization in **2**, DotWallet will redirect the client to the redirect_uri specified by the application. The authorization code code and the provided state will be included in the query parameters.
+
+4. Exchange code for access_token. The access tokens are credentials used to access protected resources, which are issued by the authorization server.
+   
+:::warning
+To avoid security issues, any request for using or obtaining `access_token` must be made from the backend server. Do not disclose your `access_token` and `client_secret`<sup>1</sup> on the client side.
+:::
+
+
+## DotWallet Developer Platform
+
+1. Before using DotWallet, you need to register and create a app on [DotWallet Developer Platform](https://developers.DotWallet.com/en).
+
+![](../../static/img/DotWallet-create-dapp.png)
+
+2. After creating the app, you will receive an email containing `app_id` and `secret`.
+
+
+![](../../static/img/DotWallet-mail.png)
+
+3. Next, you need to set [redirection URI](https://www.oauth.com/oauth2-servers/redirect-uris). Redirect URLs are a critical part of the OAuth flow. After a user successfully authorizes an application, the authorization server will redirect the user back to the application. For example, in the figure below, we filled in two redirection URIs.
+  
+- [https://dotwallet-example.vercel.app/callback/](https://dotwallet-example.vercel.app/callback/) is the redirection URI used for product deployment.
+- [http://localhost:3000/callback/](http://localhost:3000/callback/) is the redirection URI used for development and debugging.
+
+![](../../static/img/DotWallet-uris.png)
+
+
+:::note
+*Callback domain* in the form is the redirection URIs in OAuth. 
+:::
+
+## Implementation
+
+
+
+Here is a code example to create `DotWalletSigner` in [Nextjs](https://nextjs.org/):
+
+
+1. Construct URI. 
+
+```ts
+export default async function Home() {
+  const loginUrl = `https://api.ddpurse.com/authorize?client_id=${process.env.CLIENT_ID}&redirect_uri=${encodeURIComponent(
+      process.env.REDIRECT_URI || ''
+    )}&response_type=code&scope=${encodeURIComponent(
+      "user.info"
```

Comment:
Add `autopay.bsv` in scope, so app can spend users' bsv. This is not needed to show balance, but an app will almost surely need it in prod. https://developers.dotwallet.com/documents/en/#user-authorization